lib/api_constraints.rb

```diff
@@ -0,0 +1,11 @@
+# Validates the client request headers before deciding which version of the API to use
+class ApiConstraints
```

Comment:
ApiConstraints will support the Client/Server interaction and give us flexibility when we have multiple api versions